### DISCUSSION TITLE
Create Array of Object

### DISCUSSION THREAD

**Discussion by jean-Box** [mockoon/mockoon#1233] [Help]
Hello, 

I seach how can i create a array list of object and take one of it for my body response
this code return empty in value
```javascript

{{{setVar 'typesMessages' (array 
 (object type='item1')
 (object type='item2')
)}}}
{{setVar 'typeMessage' (oneOf @typesMessages)}}

{
    "typeMessage": {{@typeMessage}}
}
```

**Comment by 255kb** [comment]
Hi, what you are looking for is the `stringify` helper, otherwise your `@typeMessage` variable will get "automatically" translated to a string by Javascript which will give `[object Object]`:

```handlebars
{{setVar 'typesMessages' (array 
 (object type='item1')
 (object type='item2')
)}}      <-- we can get rid of the triple curly braces, they have no effect here
{{setVar 'typeMessage' (oneOf @typesMessages)}}

{
    "typeMessage": {{{stringify @typeMessage}}}      <-- note the triple curly braces to avoid characters encoding
}
```

You could also use the [data buckets](https://mockoon.com/docs/latest/data-buckets/overview/) which are translated to arrays/objects at runtime, and them fetch one item using `dataRaw`:

![image](https://github.com/mockoon/mockoon/assets/7489814/c4208a35-21b1-4e96-a753-5529f43b6a12)

With the following template:

```handlebars
{
    "typeMessage": {{{stringify (oneOf (dataRaw 'types'))}}}
}
```

PS: data buckets also support templating.

**Reply by tabkram** [reply]
oh. thanks for this @255kb 

>  note the triple curly braces to avoid characters encoding
